internal/framework/status/updater_test.go

```diff
@@ -508,3 +509,119 @@ var _ = Describe("Updater", func() {
 		})
 	})
 })
+
+func TestUpdateObjStatus(t *testing.T) {
+	// totalFailedAttempts: How many times the fake client fails before returning nil.
+	// allowedFailedAttempts: The total amount of times the function gets run before the
+	//						  status updater stops retrying and errors.
+	tests := []struct {
+		name                  string
+		totalFailedAttempts   int
+		allowedFailedAttempts int
+		expError              bool
+	}{
+		{
+			name:                  "fails when allowedFailedAttempts is less than totalFailedAttempts",
```

Comment:
These test names describe the test implementation rather than how the function under test is supposed to behave. 
Something like "returns an error if it times out waiting for update to succeed" or "retries until it exceeds number of attempts"

Same goes for the other test names

Reply:
Removed tests because we use imported retrying function.